pkg/backends/locust/spec.go

```diff
@@ -0,0 +1,27 @@
+package locust
+
+import (
+	"errors"
+	"time"
+
+	loadTestV1 "github.com/hellofresh/kangal/pkg/kubernetes/apis/loadtest/v1"
+)
+
+var (
+	// ErrRequireMinOneDistributedPod spec requires 1 or more DistributedPods
+	ErrRequireMinOneDistributedPod = errors.New("LoadTest must specify 1 or more DistributedPods")
+	// ErrRequireTestFile the TestFile filed is required to not be an empty string
+	ErrRequireTestFile = errors.New("LoadTest TestFile is required")
+)
+
+//BuildLoadTestSpec validates input and returns valid LoadTestSpec
+func BuildLoadTestSpec(overwrite bool, distributedPods int32, testFileStr, envVarsStr, targetURL string, duration time.Duration) (loadTestV1.LoadTestSpec, error) {
+	lt := loadTestV1.LoadTestSpec{}
+	if distributedPods <= int32(0) {
+		return lt, ErrRequireMinOneDistributedPod
+	}
+	if testFileStr == "" {
+		return lt, ErrRequireTestFile
+	}
+	return loadTestV1.NewSpec(loadTestV1.LoadTestTypeLocust, overwrite, distributedPods, testFileStr, "", envVarsStr, loadTestV1.ImageDetails{}, loadTestV1.ImageDetails{}, targetURL, duration), nil
```

Comment:
```golang
loadTestV1.ImageDetails{}, loadTestV1.ImageDetails{}
```
Should have defined image details, see:
https://github.com/hellofresh/kangal/blob/master/pkg/backends/fake/spec.go#L12
https://github.com/hellofresh/kangal/blob/master/pkg/backends/jmeter/spec.go#L27

Reply:
Instead of inserting into the Spec and persisting it, I'm using global default.
This way it will works using `kubectl apply` with yaml file even in case the image fields are left blank.

See:
https://github.com/hellofresh/kangal/blob/master/pkg/backends/locust/locust.go#L39
https://github.com/hellofresh/kangal/blob/master/pkg/backends/locust/locust.go#L193

Reply:
That was a hack before because it didn't persist specs properly, right now we build fully contained specs in the Proxy side.

Reply:
IMO the default image has to be set on the Controller, and if no image is set on the Spec, the Controller defaults should be use.
This keeps Proxy optional.

If we let the Proxy populates the Spec with image defaults, the ones set in the Controller will never be use.
And Controller would still have to check if Spec image is empty because LoadTest resource can be `kubectl` applied.

Reply:
Yes, that is an idea to build full spec LoadTest in the proxy side, so controller only takes specs and puts LoadTest to the desired state. So LoadTest spec will be a kinda immutable struct, the controller only responsible on changing statuses.

As I said that SetDefaults was a hack on the controller side because proxy didn't populate load test spec properly.

Reply:
Making `kubectl apply` for such complex CRD also not a good way (that is why we made easy for humans API on top) but for sure possible. In my opinion, if something is invalid in LoadTest spec controller should set Error state and not making any changes inside a spec.

Reply:
This is the same discussion at https://github.com/hellofresh/kangal/pull/42

I think I got your point, you are saying that Spec images are mandatory and if Controller find an empty string it should throw and error?

Reply:
That was an initial idea but it didn't work because we had a mistake on our crd.yaml and we made this hack to populate that defaults every time (basically it was never persisting those values, so each sync it was setting that defaults :-D)